File: common/src/main/java/com/study/common/security/InternalAuthConfig.java
package com.study.common.security;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class InternalAuthConfig {

    @Bean
    public InternalTokenProvider internalTokenProvider(
            @Value("${internal.auth.secret}") String secret) {
        return new InternalTokenProvider(secret);
    }
}

File: common/src/main/kotlin/com/study/common/security/InternalAuthConfig.kt
package com.study.common.security

import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.web.servlet.FilterRegistrationBean
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class InternalAuthConfig {

    @Bean
    fun internalTokenProvider(
        @Value("\${internal.auth.secret}") secret: String
    ): InternalTokenProvider = InternalTokenProvider(secret)

    @Bean
    fun internalAuthFilter(tokenProvider: InternalTokenProvider): FilterRegistrationBean<InternalAuthFilter> =
        FilterRegistrationBean(InternalAuthFilter(tokenProvider)).apply {
            addUrlPatterns("/api/*")
            order = 1
        }
}
